build_tools/benchmarks/post_benchmark_comment.py

```diff
@@ -0,0 +1,205 @@
+
+  def __init__(self, github_token: str):
+    self._api_headers = {
+        "Accept": "application/vnd.github+json",
+        "Authorization": f"token {github_token}",
+        "X-GitHub-Api-Version": GITHUB_API_VERSION,
+    }
+
+  def get(self, endpoint: str, payload: Any) -> requests.Response:
+    return requests.get(endpoint,
+                        data=json.dumps(payload),
+                        headers=self._api_headers)
+
+  def post(self, endpoint: str, payload: Any) -> requests.Response:
+    return requests.post(endpoint,
+                         data=json.dumps(payload),
+                         headers=self._api_headers)
+
+  def patch(self, endpoint: str, payload: Any) -> requests.Response:
+    return requests.patch(endpoint,
+                          data=json.dumps(payload),
+                          headers=self._api_headers)
+
+
+def post_to_gist(requester: APIRequester,
+                 filename: str,
+                 content: str,
+                 verbose: bool = False) -> str:
+  """Posts the given content to a new GitHub Gist and returns the URL to it."""
+
+  response = requester.post(endpoint=GITHUB_GIST_API,
+                            payload={
+                                "public": True,
+                                "files": {
+                                    filename: {
+                                        "content": content
+                                    }
+                                }
+                            })
+  if response.status_code != http.client.CREATED:
+    raise RuntimeError(
+        f"Failed to comment on GitHub; error code: {response.status_code}")
+
+  response = response.json()
+  if verbose:
+    print(f"Gist posting response: {response}")
+
+  if response["truncated"]:
+    raise RuntimeError(f"Content too large and gotten truncated")
+
+  return response["html_url"]
+
+
+def get_previous_comment_on_pr(requester: APIRequester,
+                               pr_number: int,
+                               gist_bot_user: str,
+                               comment_type_id: str,
+                               verbose: bool = False) -> Optional[int]:
+  """Gets the previous comment's id from GitHub."""
+
+  for page in range(1, MAX_PAGES_TO_SEARCH_PREVIOUS_COMMENT + 1):
+    response = requester.get(
+        endpoint=f"{GITHUB_IREE_API_PREFIX}/issues/{pr_number}/comments",
+        payload={
+            "per_page": 100,
+            "page": page,
+            "sort": "updated",
+            "direction": "desc"
+        })
+    if response.status_code != http.client.OK:
+      raise RuntimeError(
+          f"Failed to get PR comments from GitHub; error code: {response.status_code}"
+      )
+
+    comments = response.json()
+    if verbose:
+      print(f"Previous comment query response on page {page}: {comments}")
+
+    # Find the last comment from GITHUB_ACTIONS_USER and has the comment type id.
```

Comment:
"last" is a bit odd here. We're stopping on the *first* comment we see, but I guess "desc" in the query means it's in the opposite order. But then you also have it sorted by updated time instead of created time. So I think in the end it's the most recently updated comment from the bot.

Reply:
Updated the comment. Correct, we want to find the most recently updated comment.